lib/greed/game.rb

```diff
@@ -0,0 +1,102 @@
+absolute_path = File.expand_path(File.dirname(__FILE__))
+require absolute_path + '/turn'
+require absolute_path + '/player'
+
+module Greed
+  class Game
+
+    def initialize
+      @turn = Turn.new
+      @players = []
+    end
+
+    def new_game
+      player_count = player_count?
+      init_players(player_count)
+      get_player_names
+      play_game
+    end
+    
+    # Complete a full game of Greed, entering a final round when any player reaches
+    # a score of 3000 or more
+    def play_game
+      final_round = false
+      while not final_round
+        current_player = @players.pop
+        puts "Current player: #{current_player.name} (current score: #{current_player.score})"
+        score = @turn.take_turn
+
+        # Add score to player total if the player is 'in the game'
+        # (scored more than 300 in a single turn at least once)
+        if current_player.score > 0 || score >= 300
+          puts "Your final score for this round was #{score}"
+          current_player.score += score
+          puts "Your total score is now #{current_player.score}"
+        else
+          puts "You did not score high enough to enter the game."
+        end
+        puts # Spacing
+
+        # Check if this player has reached 3000, which triggers a final round
+        if current_player.score >= 3000 then final_round = true end
+        # Otherwise, place them at the end of the line
+        if not final_round then @players.unshift(current_player) end
```

Comment:
Our standard style conventions and to always use `!` instead of `not`.

For one line `if` statements you can simply move the expression to the start of the line, with the if conditional following.

`@players.unshift(current_player) if !final_round`